### PR TITLE
Checkout: add support for CNPJ numbers for BR customers

### PR DESCRIPTION
**Pull request by yoavf**
#### Changes proposed in this Pull Request

* Up until now we only supported individuals tax identification numbers (CPF). With this change, we now support companies tax identification numbers (CNPJ) as well.

#### Testing instructions

* With an account set to the BRL currency, add plan/product to the cart and proceed to checkout
* In the credit card form, chose Brazil as your country
* The additional form fields for Brazil should show up
* The Taxpayer identification number field should accept both `94.065.313/0001-71` and `831.836.515-13` - both typed and pasted.

Note: if your UI is in Brazillian Portuguese, the "Taxpayer identification number" field will be labeled "CPF" - this is set in the translation and will be updated once this is merged.

### CODE REVIEW

File: client/lib/checkout/validation.js [hunk 1]
```diff
@@ -204,17 +205,20 @@ validators.validExpirationDate = {
...
 		}
-		return isValidCPF( value );
+		return isValidCPF( value ) || isValidCNPJ( value );
 	},
 	error: function( description ) {
-		return i18n.translate( '%(description)s is invalid. Must be in format: 111.444.777-XX', {
-			args: { description: description },
-		} );
+		return i18n.translate(
+			'%(description)s is invalid. Must be in format: 111.444.777-XX or 11.444.777/0001-XX',
```
Comment (a8ci18n) [inline client/lib/checkout/validation.js]: :disappointed: 7 existing translations will be lost with this change.

Comment (yoavf) [reply client/lib/checkout/validation.js]: That's ok :)

Comment (akirk) [reply client/lib/checkout/validation.js]: What about we make this a variable `%(description)s is invalid. Must be in format: %(validFormat)`?

**Comment by yoavf** [comment]
> Copy and paste fails for CNPJ numbers e.g: `94065313000171` that really needs a fix.
> 
> I think removing the magic rewrite / formatting stuff is probably the best option. Its kind of jarring to use even typing a normal CPF number. If people want to add in `/`, `.` and `-` just strip them on submit.

Pinging @eduardozulian  @gugaalves @RafaelFunchal @carinapilar  - how do CPF/CNPJ fields usually behave? If pasting, would it be expected to paste just the number, or the numbers plus the separators ( `/`, `.` and `-` )?

**Comment by carinapilar** [comment]
Usually, the user would paste (or type) just numbers, and then the field will have a mask that shows the separators.

**Comment by yoavf** [comment]
I fixed the masking function - it should be much better now, and accept typing and pasting of both numbers (pasting with and without the separators).

**Comment by lsl** [comment]
Tested, copy paste works now, thanks! 

Approved changes, I'm OK with the PR being merged as is.

--

We probably want to raise an issue to split the values or remove the masking completely.

The flow for entering a cnpj formatted value is just really broken, I don't think its actually possible to do dual format masking in a user friendly way. I also noticed, backspace doubles over masked values, masking values are ignored, and edits reset cursor position to the end of the field too. There might be more than that but that's what I could nail down.

We could probably fix some of those but I don't think we'll get around the flow issue for cnpj. The cpf format is just really jarring when you expect a different format than you see.

Thoughts?

**Comment by yoavf** [comment]
@lsl  I agree we can do better - do doubt about it. Though I'll label that as "maybe-later" because I haven't heard many complaints on the existing field, and this is actually an improvement.

We'll have the opportunity to do this when we implement other "document" type fields for other countries, which is coming up soon-ish.

**Pull request merged**